android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -74,8 +76,18 @@ fun purchaseProduct(
     if (activity != null) {
         val onReceiveStoreProducts: (List<StoreProduct>) -> Unit = { storeProducts ->
             val productToBuy = storeProducts.firstOrNull {
-                // TODO: Verify this works because "subId:basePlanId" (it should but strings are silly)
-                it.id == productIdentifier && it.type.name.equals(type, ignoreCase = true)
+                // Comparison for when productIdentifier is "subId:basePlanId"
+                val foundByProductIdContainingBasePlan =
+                    (it.id == productIdentifier && it.type.name.equals(type, ignoreCase = true))
+
+                // Comparison for when productIdentifier is "subId" and googleBasePlanId is "basePlanId"
```

Comment:
Hmm can you explain in what situations we would pass a "subId:basePlanId" from the hybrids, and when a "subId" and a "basePlanId" separately?

Reply:
The `purchaseStoreProduct(StoreProduct)` is not required but it seems like it would be nice to offer up this option when making this change 🤷‍♂️

Reply:
Hmm interesting... Ok, I would mention this as a comment here then? So there is more context why both comparisons are needed.

Reply:
And yeah, I think adding a `purchasesStoreProduct` would be great (sad that javascript doesn't support overloading :( ), and maybe even deprecating the one that receives a string? Not sure if we need that for a specific reason... CC @vegaro

Reply:
the one that receives a string is from the old times when we didn't have typescript and we didn't have a Product interface.

Reply:
@vegaro Aye aye! Will add those to RN when I get back there 😊

Reply:
`purchaseStoreProduct(StoreProduct)` is going to remove the need to pass the type right? I am so excited about that change. There are so many support tickets of people that forget to pass the type...